### PR TITLE
IDP-1261 - Lib maintenance

### PR DESCRIPTION
**Pull request by Gcaya**
## Description of changes

Introduced coding standards package and fixed warnings

## Additional checks

- [ ] Updated the documentation of the project to reflect the changes
- [ ] Added new tests that cover the code changes

### CODE REVIEW

**Review by heqianwang — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -15,7 +15,7 @@ jobs:
 
       - uses: actions/setup-dotnet@v4
         with:
-          dotnet-version: "6.0.x"
+          global-json-file: "global.json"
```
Comment (heqianwang) [inline .github/workflows/ci.yml:18]: just curious why is this necessary? Can we just use:
`- uses: actions/setup-dotnet@v4`

File: src/Workleap.ComponentModel.DataAnnotations.Tests/ContainsOnlyNonEmptyGuidsAttributeTests.cs [hunk 2]
```diff
@@ -40,23 +40,23 @@ public void Validator_TryValidateObject_Returns_The_Expected_Error_Message_When_
         Assert.Equal(expectedErrorMessage, result.ErrorMessage);
     }
 
-    private class ValidData : IEnumerable<object?[]>
+    private sealed class ValidData : IEnumerable<object?[]>
```
Comment (meziantou) [inline src/Workleap.ComponentModel.DataAnnotations.Tests/ContainsOnlyNonEmptyGuidsAttributeTests.cs:43]: A better cleanup would be to use a MemberData and a TheoryData<T>

Comment (Gcaya) [reply src/Workleap.ComponentModel.DataAnnotations.Tests/ContainsOnlyNonEmptyGuidsAttributeTests.cs]: Changed to TheoryData

**Review by meziantou — APPROVED** [review]
